src/ssl.rs

```diff
@@ -233,6 +254,12 @@ impl<IOCB: IOCallbacks> Session<IOCB> {
     ///
     /// [0]: https://www.wolfssl.com/documentation/manuals/wolfssl/group__IO.html#function-wolfssl_negotiate
     pub fn try_negotiate(&self) -> PollResult<()> {
+        // SAFETY: [`wolfSSL_negotiate`][0] ([also][1]) expects valid pointer to `WOLFSSL` and since the `WOLFSSL` struct
+        // can be used in multiple threads based on [`Library design`][2], protected by a mutex lock
```

Comment:
The "and since  ... protected by a mutex lock" clause is pretty hard to grok IMHO.

How about:

> ... expects a valid pointer to a `WOLFSSL`. Per the [`Library design`][2] access is synchronized via the containing [`Mutex`]?

Reply:
Updated in all occurrences of the above statement.